go/4_auth-aws.go

```diff
@@ -25,36 +25,21 @@ func getSecretWithAWSAuthIAM() (string, error) {
 		return "", fmt.Errorf("unable to initialize Vault client: %w", err)
 	}
 
-	logger := hclog.Default()
-
-	// If environment variables are empty, will fall back on other AWS-provided mechanisms to retrieve credentials.
-	creds, err := awsutil.RetrieveCreds(os.Getenv("AWS_ACCESS_KEY_ID"), os.Getenv("AWS_SECRET_ACCESS_KEY"), os.Getenv("AWS_SESSION_TOKEN"), logger)
-	if err != nil {
-		return "", fmt.Errorf("unable to retrieve creds from STS: %w", err)
-	}
-
-	// the optional second parameter can be used to help mitigate replay attacks,
-	// when the role in Vault is configured with resolve_aws_unique_ids = true: https://www.vaultproject.io/docs/auth/aws#iam-auth-method
-	params, err := awsutil.GenerateLoginData(creds, "Replace-With-IAM-Server-Id", os.Getenv("AWS_DEFAULT_REGION"), logger)
+	awsAuth, err := auth.NewAWSAuth(
```

Comment:
I'm realizing this is also called `awsAuth`, I'm not sure if there's a difference in Go with the naming conventions for libraries vs. regular variables

Reply:
Nope, no difference between the naming conventions that I'm aware of.